### ISSUE TITLE
respect-bl still delete blank lines from dependencies

### ISSUE THREAD

**Issue by Villar74** [kkinnear/zprint#132]
Here is my config
```
:zprint
                    {:style [:respect-nl :respect-bl]
                     :width 100
                     :parallel? true
                     :fn-map
                     {"reg-event-fx" :arg1-force-nl
                      "reg-event-db" :arg1-force-nl
                      "reg-sub" :arg1-force-nl
                      "reg-fx" :arg1-force-nl
                      "if" :arg1-force-nl}
                     :map
                     {:comma? false
                      :hang? false
                      :sort? true
                      :force-nl? true}
                     :vector {:respect-nl? true
                              :respect-bl? true
                              }
                     :old? false}
```

and in project.clj zprint still delete blank lines like this
```
:dependencies [;; CLOJURE

                 [org.clojure/clojure "1.10.1"]
                 ;; Library for async programming.
                 [org.clojure/core.async "0.4.474"]
                 ;; Pattern matching library.
                 [org.clojure/core.match "0.3.0"]]
```
=>
```
:dependencies [;; CLOJURE
                 [org.clojure/clojure "1.10.1"]
                 ;; Library for async programming.
                 [org.clojure/core.async "0.4.474"]
                 ;; Pattern matching library.
                 [org.clojure/core.match "0.3.0"]]
```

**Comment by kkinnear** [comment]
Wow, that's pretty broken.  I can reproduce it with no problem.  I'll fix it directly, thanks for noticing and reporting this!

A question:  While I __can__ reproduce the loss of the blank line, I __cannot__ reproduce the alignment you show for the `[;; CLOJURE` and the line below it.  It is possible that when you cut and pasted the output into the issue that the line with `:dependencies [;; CLOJURE` got shifted two chars to the left?

I think you might be the first person to actually include their configuration in the issue, so thanks for that too!  Since you did that, I noticed that you had selected both `:respect-nl` and `:respect-bl` as styles.  I never imagined that anyone would do that!  

As far as I know, `:respect-nl` is what is actually taking precedence (despite being first in the vector of styles), since if they are both specified what actually takes precedence is which one I look for first in the code.   And I'm pretty sure that is always `:respect-nl?` in the detailed configuration for `:list`, `:map`, `:set`, and `:vector`.  Not that you necessarily care about all of this.  The point is that I expected that people would pick __either__ `:respect-nl` __or__ `:respect-bl`, not both.  I will think about this some more.  After I fix this bug, of course.

**Comment by Villar74** [comment]
Yes, that's possible. Using :respect-nl or :respect-bl was the last try to fix this bug, firstly I tried to add one of them

**Comment by kkinnear** [comment]
This was actually quite an involved problem.  It is fixed in 1.0.0.  Thanks for noticing and submitting it!
